### PR TITLE
add .cb_get_samples_table_v2

### PR DESCRIPTION
**Pull request by ilevantis**
## This PR does the following:

+ It adds the function `.cb_get_samples_table_v2()` and renames the old `.cb_get_samples_table()` to `.cb_get_samples_table_v1()`.
+ The exported function ` cb_get_samples_table()` now calls `.cb_get_samples_table_v1()` or `.cb_get_samples_table_v2()` depending on the CB version of the cohort it is given.

## Testing

Get the package from the PR branch:
```shell
> git clone 'https://github.com/lifebit-ai/cloudos.git'                                                                                                                                   
> cd cloudos
> git checkout cb_get_samples_table_v2
```

In the cloudos directory enter an R session (or do so in Rstudio) and load the package + config:
```R
> devtools::install(".")
> library(cloudos)
> cloudos_configure(base_url = "http://cohort-browser-dev-110043291.eu-west-1.elb.amazonaws.com/cohort-browser/", 
token = "...api token...",
team_id = "5f7c8696d6ea46288645a89f")
```

Test that you are able to load the cohorts in v1 and v2:

```R
> cohortv1 <- cb_load_cohort("60dedc0df97e550eda221fed", cb_version = "v1")
> cb_get_samples_table(cohortv1)
       EID    Sex Maternal age during pregnancy (years) Abnormal antenatal findings During pregnancy Perinatal complications
1  1000622 Cancer                         West Midlands                        1954   White: British                      NA
2  1001391 Cancer                         West Midlands                        1955   White: British                      NA
3  1001472 Cancer                          North Thames                        1953       Not Stated                      NA
4  1001532 Cancer             Genomics Network Alliance                        1953        Not Known                      NA
5  1001749 Cancer                           West London                        1965        Not Known                      NA
6  1001967 Cancer                         West Midlands                        1943   White: British                      NA
7  1002061 Cancer                       East of England                        1958       Not Stated                      NA
8  1002114 Cancer                       West of England                        1977     White: Irish                      NA
9  1002405 Cancer                       East of England                        1970       Not Stated                      NA
10 1002451 Cancer                                Oxford                        1932       Not Stated                      NA
   Method of diagnosis
1                   NA
2                   NA
3                   NA
4                   NA
5                   NA
6                   NA
7                   NA
8                   NA
9                   NA
10                  NA

> cohortv2 <- cb_load_cohort("60dedc0df97e550eda221fed", cb_version = "v2")
> cb_get_samples_table(cohortv2)
       EID           Sex Maternal age during pregnancy (years) Abnormal antenatal findings                  During pregnancy
1  1000289 Rare Diseases                          North Thames                        1950                    White: British
2  1000522 Rare Diseases                  Yorkshire and Humber                        1990 White: Any other White background
3   100079 Rare Diseases             Genomics Network Alliance                        1967                    White: British
4  1001937 Rare Diseases                         West Midlands                        1954                      White: Irish
5  1002061        Cancer                       East of England                        1958                        Not Stated
6  1002114        Cancer                       West of England                        1977                      White: Irish
7  1002451        Cancer                                Oxford                        1932                        Not Stated
8  1003149 Rare Diseases                          North Thames                        1986                    White: British
9  1003511        Cancer                       East of England                        1972                    White: British
10 1003836 Rare Diseases                         West Midlands                        1952 Asian or Asian British: Pakistani
   Perinatal complications Method of diagnosis
1             Not Supplied             Proband
2             Not Supplied            Relative
3             Not Supplied            Relative
4             Not Supplied             Proband
5                       NA                  NA
6                       NA                  NA
7                       NA                  NA
8             Not Supplied            Relative
9                       NA                  NA
10            Not Supplied             Proband
```

### CODE REVIEW

File: R/cb_cohort_extract.R [hunk 1]
```diff
@@ -141,36 +156,118 @@ cb_get_samples_table <- function(cohort,
...
+  res_df <- dplyr::bind_rows(df_list)[-1,] # combine and remove empty first row
+  
   
   # check if the dataframe is retrieved properly
   if(length(res_df) == 0){
     stop("Couldn't able to retrive the dataframe, something wrong with the cohort filters.")
   }
   
+  # rename the dataframe with column names
+  colnames(res_df) <- col_names
```
Comment (sk-sahu) [inline R/cb_cohort_extract.R:186]: For some reason the column names doesn't match with the contents of it, and I think this is becasue of how we are forming the `col_names` object. Can you please have a look 

```R
cohortv2 <- cb_load_cohort("60dedc0df97e550eda221fed", cb_version = "v2")
cb_get_samples_table(cohortv2)
        EID           Sex Maternal age during pregnancy (years) Abnormal antenatal findings                  During pregnancy
1  1000289 Rare Diseases                          North Thames                        1950                    White: British
2  1000522 Rare Diseases                  Yorkshire and Humber                        1990 White: Any other White background
3   100079 Rare Diseases             Genomics Network Alliance                        1967                    White: British
4  1001937 Rare Diseases                         West Midlands                        1954                      White: Irish
5  1002061        Cancer                       East of England                        1958                        Not Stated
6  1002114        Cancer                       West of England                        1977                      White: Irish
7  1002451        Cancer                                Oxford                        1932                        Not Stated
8  1003149 Rare Diseases                          North Thames                        1986                    White: British
9  1003511        Cancer                       East of England                        1972                    White: British
10 1003836 Rare Diseases                         West Midlands                        1952 Asian or Asian British: Pakistani
   Perinatal complications Method of diagnosis
1             Not Supplied             Proband
2             Not Supplied            Relative
3             Not Supplied            Relative
4             Not Supplied             Proband
5                       NA                  NA
6                       NA                  NA
7                       NA                  NA
8             Not Supplied            Relative
9                       NA                  NA
10            Not Supplied             Proband
```

**Review by sk-sahu — COMMENTED** [review]
Please check the output dataframe for column names and their contents, they do not match. Mentioned below where they might be getting wrong

File: R/cb_cohort_extract.R (hunk 1 above)
Comment (ilevantis) [reply R/cb_cohort_extract.R:186]: The names are actually correct. The data in the cohort browser instance is just mislabelled. The web UI shows the same values for those columns.

Comment (sk-sahu) [reply R/cb_cohort_extract.R:186]: Ah okay, just checked. Yes, then it makes sense. Thank you for clarification

File: R/cb_cohort_extract.R [hunk 2]
```diff
@@ -113,6 +113,21 @@ cb_get_genotypic_table <- function(cohort,
 cb_get_samples_table <- function(cohort,
                               page_number = 0,
                               page_size = 10) {
+  
+  if (cohort@cb_version == "v1") {
+    return(.cb_get_samples_table_v1(cohort))
+    
+  } else if (cohort@cb_version == "v2") {
+    return(.cb_get_samples_table_v2(cohort))
```
Comment (sk-sahu) [inline R/cb_cohort_extract.R:121]: I don't see `page_number` and `page_size` is getting to pass to internal functions. Any reason to it?

File: R/cb_cohort_extract.R [hunk 3]
```diff
@@ -113,6 +113,21 @@ cb_get_genotypic_table <- function(cohort,
 cb_get_samples_table <- function(cohort,
```
Comment (sk-sahu) [inline R/cb_cohort_extract.R:113]: `cb_version` option is missing in function definition and as well as its roxygen docs

**Review by sk-sahu — CHANGES_REQUESTED** [review]
Great Job on this PR :mechanical_arm: Awesome idea to get the cb_version from cohort object. Really liked this 100 trophy :100: :trophy: 

Few comments below, before we ready to merge

File: R/cb_cohort_extract.R (hunk 2 above)
Comment (ilevantis) [reply R/cb_cohort_extract.R]: Ah good catch. Have fixed this in a new commit.

**Review by sk-sahu — APPROVED** [review]
LGTM :rocket:

**Pull request merged**